src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -450,34 +496,53 @@ std::unique_ptr<model_spec> activity_classifier::init_model() const
   return result;
 }
 
-void activity_classifier::init_train(
-    gl_sframe data, std::string target_column_name,
-    std::string session_id_column_name, variant_type validation_data,
-    std::map<std::string, flexible_type> opts)
-{
-
-  // Begin printing progress.
-  // TODO: Make progress printing optional.
+std::tuple<gl_sframe, gl_sframe>
+activity_classifier::init_data(gl_sframe data, variant_type validation_data,
+                               std::string session_id_column_name) const {
+  gl_sframe train_data;
+  gl_sframe val_data;
   if (variant_is<gl_sframe>(validation_data)) {
-    gl_sframe validation_sf = variant_get_value<gl_sframe>(validation_data);
-    if (!validation_sf.empty()) {
-      training_table_printer_.reset(
-          new table_printer({{"Iteration", 12},
-                             {"Train Accuracy", 12},
-                             {"Train Loss", 12},
-                             {"Validation Accuracy", 12},
-                             {"Validation Loss", 12},
-                             {"Elapsed Time", 12}}));
+    train_data = data;
+    val_data = variant_get_value<gl_sframe>(validation_data);
+    if (!val_data.empty()) {
     } else {
       log_and_throw("Input SFrame either has no rows or no columns. A "
                     "non-empty SFrame is required");
     }
+  }
+  else if ((variant_is<flex_string>(validation_data)) && (variant_get_value<flex_string>(validation_data)=="auto")) {
+    gl_sarray unique_session = data[session_id_column_name].unique();
+    if (unique_session.size() > 50 && unique_session.size() < 200) {
+      // TODO: Expose seed parameter publicly. 
+      std::tie(train_data, val_data) = random_split_by_session(data, session_id_column_name, 0.9, 1);
+    }
+    else if (unique_session.size() > 200) {
```

Comment:
We may as well match the `supervised_learning` implementation exactly, which uses `>= 200`. Also, if we handle this case first, then the other case can just be `>= 50` instead of `>= 50 && < 200`